source/solvers/gls_nitsche_navier_stokes.cc

```diff
@@ -104,8 +104,9 @@ GLSNitscheNavierStokesSolver<dim, spacedim>::assemble_nitsche_restriction()
           else if (dim == 3)
             h_cell = pow(6 * cell->measure() / M_PI, 1. / 3.) /
                      this->velocity_fem_degree;
-          const double penalty_parameter = 1. / (h_cell * h_cell);
-          const auto & dh_cell =
+          const double penalty_parameter =
+            1. / std::pow(h_cell * h_cell, double(dim) / double(spacedim));
```

Comment:
would you have a reference for this?

Reply:
So previously the penalty parameter is scaled as 1/h^2. But the issue is that in dim==2 spacedim==3, it is integrated over an area. So the penalty is extremely strong. It's what made the solver very unstable when the mesh was being refined.
Here we regularize the penalty parameter so that it is scaled correctly. We have a larger error, but the convergence is recovered correctly. It's kind-off a tradeoff. I don't have a reference because we are kinda inventing this... :)